### PR TITLE
Adding new SignPostCheck and IntersectingBuildingsCheck. 

### PR DESCRIPTION
**Pull request by mgcuthbert**
Also updating MapRoulette functions to allow the use of tags when creating challenges. Updating documents to describe how to develop and debug Atlas-Checks.

SignPostCheck will check is to help identify segments that are missing the proper tagging for sign posts. 

IntersectingBuildingsCheck will, much as the name indicates, flag buildings that intersect with each other.

The documentation includes how to debug Checks that you create, as well has help for writing unit tests for you your newly created Checks.

### CODE REVIEW

**Comment by matthieun** [comment]
@mgcuthbert Can you fix the title and comment that are overlapping? Thanks!

File: docs/unit_tests.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        // Another common scenario would be to loop through the results and check something on it.
+        this.verifier.verify(flag -> {
+            // Check the flag object for some expected values
+        });
+    }   
+}
+```
+
+A basic Atlas-Checks unit test contains 3 things:
+1. A TestRule class that is essentially and atlas or multiple atlas to test against.
```
Comment (MikeGost) [inline docs/unit_tests.md]: Nit: and --> an

File: docs/unit_tests.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+2. A `ConsumerBasedExpectedCheckVerifier` instance that will execute the check you are testing against the test atlas.
+3. A series of units test using the JUnit framework.
```
Comment (MikeGost) [inline docs/unit_tests.md]: units test --> unit tests

File: docs/unit_tests.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+2. verify functions - The verify functions will be used to verify the results of the running the check over the test atlas file. Below are some useful functions
+    
+    - `verifyNotEmpty()` - Verifies that at least 1 flag was produced by the check
+    - `verifyEmpty()` - Verifies that no flags were produced by the check
+    - `verifyExpectedSize(int)` - Verifies that a specific number of flags were produced by the check
+    - `verify(Consumer<List<CheckFlag>>)` - A custom verifier allowing the developer to loop through each individual flag that is produced and verify it in any way that the developer wishes.
+    
+ ### The `TestRule`
+ 
+ As mentioned previously the TestRule is a class that is associated with the Unit test class and contains all the test data, which would essentially be an Atlas in some form or another. An Atlas can be created in 2 primary ways:
```
Comment (MikeGost) [inline docs/unit_tests.md]: There are more now! We can create an Atlas from an OSM file (either from JOSM or downloaded from openstreetmap). Consider adding these options here.

Comment (MikeGost) [reply docs/unit_tests.md]: There is a task to revamp the atlas documentation around testing, once that's done, we can link it here.

File: docs/unit_tests.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+ ```java
+@TestAtlas(nodes = { @Node(coordinates = @Loc(value = "37.32544,-122.033948")),
+            @Node(coordinates = @Loc(value = "37.33531,-122.009566")),
+            @Node(coordinates = @Loc(value = "37.3314171,-122.0304871")) }, edges = {
+                    @Edge(coordinates = { @Loc(value = "37.32544,-122.033948"), @Loc(value = "37.33531,-122.009566") }, tags = {
+                            "highway=SECONDARY" }),
+                    @Edge(coordinates = { @Loc(value = "37.33531,-122.009566"), @Loc(value = "37.3314171,-122.0304871") }, tags = {
+                            "highway=SECONDARY" }) })
+```
+- Atlas Text Files - It is fairly easy to produce very small test atlas text files which that can be stored in resources much like the [poolsize.altas](../src/test/resources/org/openstreetmap/atlas/checks/validation/areas/poolsize.atlas). And generating this files are fairly straight forward, and can be done by following these steps:
```
Comment (MikeGost) [inline docs/unit_tests.md]: Small note - we usually avoid zipping these up, and store in raw .txt format so it's easier to see what's in the atlas file with the naked eye. Might be worth pointing this out, and emphasis on small test file :)

File: docs/unit_tests.md (hunk 3 above)
Comment (mgcuthbert) [reply docs/unit_tests.md]: So yes I considered adding all the options, however I focused on the 2 that I thought might be the easiest (I might be wrong about that assessment), and then once that documentation is updated we can link directly to it for more detailed information around it.

File: docs/unit_tests.md (hunk 4 above)
Comment (mgcuthbert) [reply docs/unit_tests.md]: So all my documentation essentially ignores the option to zip any of it, for this reason. But good point to specifically point it out and make note that these files should be small.

File: docs/unit_tests.md (hunk 3 above)
Comment (MikeGost) [reply docs/unit_tests.md]: Fair enough. You could argue that an OSM file is easier to create than a text file. Text Atlases are usually the piece of interest from a larger Atlas or if created from scratch, you need to have prior understanding of the text atlas format. OSM files can be downloaded right from OSM or quickly drawn/downloaded in JOSM and saved. Most power users would probably use this over a text atlas. Either one would work, only point was mentioning them at least might help some users.

File: docs/debugging.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+The first thing required is to import the project into Intellij. Intellij supports gradle projects which is what Atlas-Checks is, so importing the project is fairly straight forward. These instructions assumed 
+
+1. Clone the Atlas-Checks project into a directory of your choice 
+2. Click on "File" -> "New" -> "Project from Existing Sources..."
+3. Select the folder that you original cloned your Atlas-Checks project into in Step 1.
+4. Select the "Import project from external model" and then select the "Gradle" option
+5. In the final screen check the box "Use auto-import" and make sure that "Use default gradle wrapper (recommended)" is selected.
+6. Click Finish and wait for gradle tasks to complete.
+
+<<<<<<< HEAD
```
Comment (mertemin) [inline docs/debugging.md]: Please remove the git merge artifacts.

File: docs/debugging.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+This will create a new Atlas-Checks project that you can start developing in Intellij. Most other full fledged IDE's should following a similar path. Smaller lightweight IDEs will simply have you point to a directory.
+=======
```
Comment (mertemin) [inline docs/debugging.md]: Same

File: docs/debugging.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+This will create a new Atlas-Checks project that you can start developing in Intellij. Most other full fledged IDEs should following a similar path. Smaller lightweight IDEs will simply have you point to a directory.
+>>>>>>> internal/apple
```
Comment (mertemin) [inline docs/debugging.md]: Same

File: docs/unit_tests.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+        // Another common scenario would be to loop through the results and check something on it.
+        this.verifier.verify(flag -> {
+            // Check the flag object for some expected values
+        });
+    }   
+}
+```
+
+A basic Atlas-Checks unit test contains 3 things:
+<<<<<<< HEAD
```
Comment (mertemin) [inline docs/unit_tests.md]: Same

File: docs/unit_tests.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+    - Program Arguments: atlas-with-this-entity<br/>
+                         -input=`[Input location for atlas file]`<br/>
+                         -osmid=`[OSM ID of primary feature]`<br/>
+                         -expand=`[Amount you want to expand around feature]`<br/>
+                         -text-output=`[Output location for resultant atlas text file]`<br/>
+    - Working directory: [Root directory of Atlas-Checks project]                     
+    - Use classpath of module: atlas-checks_main
+3. Run new AtlasTestGenerator configuration in Intellij.
+
+<<<<<<< HEAD
```
Comment (mertemin) [inline docs/unit_tests.md]: Same

**Pull request merged**